cmd/genji/restore_test.go

```diff
@@ -0,0 +1,122 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"os"
+	"testing"
+
+	"github.com/dgraph-io/badger/v3"
+	"github.com/genjidb/genji"
+	"github.com/genjidb/genji/cmd/genji/shell"
+	"github.com/genjidb/genji/document"
+	"github.com/genjidb/genji/engine/badgerengine"
+	"github.com/stretchr/testify/require"
+)
+
+func TestExecuteRestore(t *testing.T) {
+	tests := []struct {
+		name   string
+		engine string
+		path   string
+		target string
+		fail   bool
+	}{
+		{"without engine", "", "", "", true},
+		{"error with bad engine", "test", "", "", true},
+		{"error without db path", "bolt", "", "", true},
+		{"with bolt", "bolt", "test.db", "restored.db", false},
```

Comment:
We should use a temporary directory for this, using ioutil.TempDir. https://golang.org/pkg/io/ioutil/#TempDir

Here's an example with Bolt: https://github.com/genjidb/genji/blob/main/engine/boltengine/engine_test.go#L37-L42